### ISSUE TITLE
After update to latest version middleware in middleware groups breaks with Class doesn't exist

### ISSUE THREAD

**Issue by Elena0308** [laravel/framework#22033]
- Laravel Version: 5.5.20
- PHP Version: 7.1
- Database Driver & Version:

### Description:
We had the following statement in our App\Http\Kernel
   `
 protected $middlewareGroups = [
       ....
        'api' => [
            'throttle:60,1',
            'bindings',
            \App\Http\Middleware\MiddlewareName::class,
        ],
];
`
And it was working perfectly well.
After the latest update from v5.5.19 to v5.5.20 we got the following 
![elena_bug_5_5_20](https://user-images.githubusercontent.com/6377654/32612234-0aa9e126-c568-11e7-9369-d5dd21d7bc6e.png)

We fixed it by declaring the middleware in $routeMiddleware and than assigning it to the api middleware group.
### Steps To Reproduce:

1. Set Laravel/framework version to 5.5.19
2. Create custom middleware and assign it to the api middleware group with it's full class path
3. Check that everything is being executed
4. Update to v5.5.20
5. Same code not working anymore

**Comment by themsaid** [comment]
Works fine, probably something in your code but you're not sharing enough information, please share the entire exception you get on a forum thread for people to help.

I tested using:

```
'api' => [
    'throttle:60,1',
    'bindings',
    TestMiddleware::class
],
```

All fine.

**Comment by themsaid** [comment]
Also I suggest that you spin a fresh 5.5.20 app really quick and test with a dummy middleware.

**Comment by Elena0308** [comment]
So it turns out that we had something in our RouteServiceProvider


```
        Route::prefix('api')
             ->middleware('api', 'middleware')
             ->namespace($this->namespace)
             ->group(base_path('routes/api.php'));
```

Which was not throwing any error previously. So I guess this was a fix of a bug that you did from 5.5.19 to 5.5.20 which caused the error output.
